### PR TITLE
Feature/reservation form

### LINKED ISSUE #13: [3 pt] Create a reservation form - Member 4
Create a reservation form that allows the user to reserve a specified doctor. The form must require:
- Doctor (if the user was redirected from the _reserve_ page).
- Date
- City

### PR DESCRIPTION
**Pull request by Camilovelag**
PR to close #13  💯 

### TBD ☑️ 
- Improve styling.
- Change the booked_appointment alert message method.
- Truncate dropdown select lists for `cities` and `doctors` fields (show only 5 options and add scroll when expanded).
- Update validators for form fields.

I'll be waiting for your feedback. Happy coding 👋

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ function App() {
       <Routes>
         <Route path="/" element={<Doctors />} />
         <Route path="/add_doctor" element={<AddDoctor />} />
-        <Route path="/add_appointment/:id?" element={<AddReservation />} />
+        <Route path="/add_appointment/:doctorId" element={<AddReservation />} />
```
Comment (asdt560) [inline src/App.js:19]: Why did you make this change? I believe the code `:id?` was there to allow for the route to pass along information but also allow for it to be sent without a number.

**Review by asdt560 — COMMENTED** [review]
Good job @Camilovelag! I do have a few comments, or rather questions for you before I approve this though. I know Jonathan worked on that aspect of the pages from the other end so it might be something to talk about with him as well!

**Review by JonahKayizzi — DISMISSED** [review]
### Great work @Camilovelag 🏁 

The Reservation form is well implemented. Please merge

File: src/components/NavPanel.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const NavPanel = () => {
         </NavLink>
         <NavLink
           className={({ isActive }) => (isActive ? activeClassName : inactive)}
-          to="/add_appointment"
+          to="/add_appointment/0"
```
[the remark this answers is not in the document]
Comment (Camilovelag) [reply src/components/NavPanel.js:19]: '0' (zero) covers the case if the reservation form is accessed from NavPanel. I considered changing the value to 'new' instead.

File: src/App.js (hunk 1 above)
Comment (Camilovelag) [reply src/App.js:19]: To differentiate this `:id` from:
 https://github.com/JonahKayizzi/doctors-reservations-frontend/blob/a766b7ca8eee7e8b68329569646f7a9fa251658d/src/App.js#L22
into `params` and avoid unexpected behaviors, but probably it's also unnecessary.

File: src/components/NavPanel.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Camilovelag) [reply src/components/NavPanel.js:19]: I passed a '0' value to enable the `doctor_field` in the form when a user accessed the form from NavPanel, but I could configure the logic to work using this approach.

[the remark this answers is not in the document]
Comment (Camilovelag) [reply src/components/NavPanel.js]: Done 😄

File: src/App.js (hunk 1 above)
Comment (Camilovelag) [reply src/App.js]: Done 😄

**Review by cvalencia1991 — COMMENTED** [review]
## Hi @Camilovelag, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Excellent Implementation

### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------

**Pull request merged**